multivariate_grouped_data_plot.Rmd

```diff
@@ -0,0 +1,825 @@
+
+- color, size, linetype：边框线的颜色，大小和线型。    
+
+- fill：箱形图区域填充颜色。     
+
+- outlier.colour, outlier.shape, outlier.size：异常值点的颜色，形状和大小。   
+
+#### 3.2.1 创建基本箱形图   
+   
+(1) 标准箱形图和缺口箱形图
+
+```{r}
+# Default plot
+e <- ggplot(ToothGrowth, aes(x = dose, y = len))
+e + geom_boxplot()
+# Notched box plot with mean points
+e + geom_boxplot(notch = TRUE, fill = "lightgray")+
+  stat_summary(fun.y = mean, geom = "point",
+               shape = 18, size = 2.5, color = "#FC4E07")
+```
+   
+(2) 按组更改箱形图颜色
+
+```{r}
+# Color by group (dose)
+e + geom_boxplot(aes(color = dose))+
+  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07"))
+# Change fill color by group (dose)
+e + geom_boxplot(aes(fill = dose)) +
+  scale_fill_manual(values = c("#00AFBB", "#E7B800", "#FC4E07"))
+```
+
+请注意，可以将scale_x_discrete()函数用于：  
+
+- 选择要显示的项目：例如c("0.5","2")，
+
+- 更改项目的顺序：例如从c("0.5","1","2")更改为c("2","0.5","1")  
+
+例如，键入：
+
+```{r}
+# Choose which items to display: group "0.5" and "2"
+e + geom_boxplot() + 
+  scale_x_discrete(limits=c("0.5", "2"))
+# Change the default order of items
+e + geom_boxplot() +
+  scale_x_discrete(limits=c("2", "0.5", "1"))
+```
+
+#### 3.2.2 创建具有多个组的箱形图
+
+使用两个不同的分组变量：dose作为x轴变量，supp作为填充颜色（图例变量）。 
+
+使用函数position_dodge()可以调整分组箱形图之间的间隔。
+
+```{r}
+e2 <- e + geom_boxplot(
+  aes(fill = supp),
+  position = position_dodge(0.9) 
+  ) +
+  scale_fill_manual(values = c("#999999", "#E69F00"))
+e2
+```
+
+使用函数facet_wrap()可将图分成多个面板：
+
+```{r}
+e2 + facet_wrap(~supp)
+```
+
+### 3.3 小提琴图
+
+小提琴图类似于箱形图，不同之处在于它们还显示了不同值的数据的核概率密度。
+
+通常，小提琴图将包括数据中位数的标记和指示四分位数范围的框，如在标准框图中一样。       
+
+关键函数：
+
+(1) geom_violin()：创建小提琴图。主要参数：
+
+- color, size, linetype：边框线的颜色，大小和线型。 
+
+- fill：区域填充颜色。
+
+- trim：逻辑变量。如果为TRUE(默认)，则将小提琴的尾部修整到数据范围。如果为FALSE，则不修剪尾部数据。
+
+(2) stat_summary()：在小提琴图上添加摘要统计信息（平均值，中位数等）。
+
+#### 3.3.1 使用摘要统计信息创建基本的小提琴图
+
+```{r}
+# Add mean points +/- SD
+# Use geom = "pointrange" or geom = "crossbar"
+e + geom_violin(trim = FALSE) + 
+  stat_summary(
+    fun.data = "mean_sdl",  fun.args = list(mult = 1), 
+    geom = "pointrange", color = "black"
+    )
+    
+# Combine with box plot to add median and quartiles
+# Change color by groups
+e + geom_violin(aes(fill = dose), trim = FALSE) + 
+  geom_boxplot(width = 0.2)+
+  scale_fill_manual(values = c("#00AFBB", "#E7B800", "#FC4E07"))+
+  theme(legend.position = "none")
+```
+
+#### 3.3.2 创建具有多个组的小提琴图
+
+```{r}
+e + geom_violin(
+  aes(color = supp), trim = FALSE,
+  position = position_dodge(0.9) 
+  ) +
+  geom_boxplot(
+    aes(color = supp), width = 0.15,
+    position = position_dodge(0.9)
+    ) +
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+```
+
+
+### 3.4 点图
+
+关键函数：
+
+geom_dotplot()：创建堆叠的点，每个点代表一个观测值。主要参数：
+
+- stackdir：点堆叠的方向。 “上”（默认），“下”，“中心”，“整个中心”（居中，但点对齐）。
+
+- stackratio：堆叠点的距离。默认值为1，其中点与点之间仅接触。对于较小的重叠点，请使用较小的值。
+
+- olor, fill：点边界颜色和区域填充。
+
+- dotsize：相对于binwidth的点的直径，默认为1。
+
+对于小提琴图，通常将摘要统计信息添加到点图。
+
+#### 3.4.1 创建基本点图
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.